### PR TITLE
testifylint: enable bool-compare rule

### PR DESCRIPTION
**Pull request by mmorel-35**
Thank you for contributing to Harbor!

# Comprehensive Summary of your change

Introduces [testifylint](https://github.com/Antonboom/testifylint) linter to help use best practices with [testify](https://github.com/stretchr/testify) library
As it can be modifying a lot of lines this only fixes one rule at a time, here is bool-compare

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [x] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

**Comment by wy65701436** [comment]
Thanks, @mmorel-35, for your contribution. I'm not familiar with testifylint and am unsure if its rules are industry standards. Could you provide more information on this?

I'm concerned because we previously encountered issues with gocilint, such as compatibility problems with different Go versions. While new tools can be beneficial, we need to consider the maintenance effort required.

File: src/controller/proxy/local_test.go [hunk 1]
```diff
@@ -85,7 +85,7 @@ func (lh *localHelperTestSuite) TestBlobExist_False() {
 	lh.registryClient.On("BlobExist", mock.Anything, mock.Anything).Return(false, nil)
 	exist, err := lh.local.BlobExist(ctx, art)
 	lh.Require().Nil(err)
-	lh.Assert().Equal(false, exist)
```
Comment (Vad1mo) [inline src/controller/proxy/local_test.go:88]: I like it, It spotted already the typical error with assertions and boolean logic.

```
	lh.Assert().Equal(false, exist)
	lh.Assert().False(exist)
```

@mmorel-35 is the only thing it finds

**Review by Vad1mo — APPROVED** [review]

**Comment by Vad1mo** [comment]
We had a discussion regarding these PRs among the community and maintainers. 
While it adds  some best practice additional complexity and barrier on new PRs does not justify this overhead.

hence we are closing this

**Comment by OrlinVasilev** [comment]
@mmorel-35 - here is the discussion in our last community meeting https://youtu.be/khVgeXJzfJo

**Pull request closed without merging**